modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/AnnotationProcessorUtils.java

```diff
@@ -0,0 +1,192 @@
+final class AnnotationProcessorUtils
+{
+
+    private AnnotationProcessorUtils()
+    {
+    }
+
+    static Map<String, VariableElement> getFieldsAnnotatedWith(TypeElement typeElement, Class<? extends Annotation> annotation)
+    {
+        ImmutableMap.Builder<String, VariableElement> fields = ImmutableMap.builder();
+
+        for (VariableElement variableElement : ElementFilter.fieldsIn(typeElement.getEnclosedElements()))
+        {
+            if (variableElement.getAnnotation(annotation) != null)
+            {
+                fields.put(variableElement.getSimpleName().toString(), variableElement);
+            }
+        }
+
+        return fields.build();
+    }
+
+    static Map<String, ExecutableElement> getMethodsAnnotatedWith(TypeElement typeElement, Class<? extends Annotation> annotation)
+    {
+        ImmutableMap.Builder<String, ExecutableElement> fields = ImmutableMap.builder();
+
+        for (ExecutableElement executableElement : ElementFilter.methodsIn(typeElement.getEnclosedElements()))
+        {
+            if (executableElement.getAnnotation(annotation) != null)
+            {
+                fields.put(executableElement.getSimpleName().toString(), executableElement);
+            }
+        }
+
+        return fields.build();
+    }
+
+
+    static MethodDocumentation getMethodDocumentation(ProcessingEnvironment processingEnv, Element element)
+    {
+        final StringBuilder parsedComment = new StringBuilder();
+        final ImmutableMap.Builder<String, String> parameters = ImmutableMap.builder();
+        parseJavaDoc(processingEnv, element, new JavadocParseHandler()
+        {
+            @Override
+            void onParam(String param)
+            {
+                parseMethodParameter(parameters, param);
+            }
+
+            @Override
+            void onBodyLine(String bodyLine)
+            {
+                parsedComment.append(bodyLine).append('\n');
+            }
+        });
+
+        return new MethodDocumentation(stripTags(parsedComment.toString()), parameters.build());
+    }
+
+    static String getJavaDocSummary(ProcessingEnvironment processingEnv, Element element)
+    {
+        final StringBuilder parsedComment = new StringBuilder();
+        parseJavaDoc(processingEnv, element, new JavadocParseHandler()
+        {
+            @Override
+            void onParam(String param)
+            {
+            }
+
+            @Override
+            void onBodyLine(String bodyLine)
+            {
+                parsedComment.append(bodyLine).append('\n');
+            }
+        });
+
+        return stripTags(parsedComment.toString());
+    }
+
+    private static String stripTags(String comment)
+    {
+        StringBuilder builder = new StringBuilder();
+        boolean insideTag = false;
+        comment = comment.trim();
+
+        for (int i = 0; i < comment.length(); i++)
+        {
+            if (comment.charAt(i) == '{' &&
+                comment.charAt(i + 1) == '@')
```

Comment:
I guess this would throw an exception when the index corresponds to the char in the comment